### DISCUSSION TITLE
How to extend flash.display.Stage?

### DISCUSSION THREAD

**Discussion by ylazy** [airsdk/Adobe-Runtime-Support#2558] [General]
Hi everyone!
Has anyone ever tried to extend the Stage class?
When I do that, I got this error:
```as
ArgumentError: Error #2012: SmartStage$ class cannot be instantiated.
```
I see that Stage is not marked as a final class. So I didn't understand why I got this error.

Any suggestions would be appreciated!
Thanks!

**Comment by ajwfrost** [comment]
Interesting .. yes, not marked as final, and yet it's set up as a "restricted" class (like a whole bunch of others such as Capabilities, Worker, Program3D, Security etc) which means they cannot be directly instantiated.

A Stage is meant to represent the visible area of a display list, and that's part of the framework that the AIR runtime provides. So the runtime can create a stage, but you can't manually create one. I would suggest your class needs to encapsulate the stage, rather than try to derive from it - there's no capability for the 'stage' objects to be anything other than the 'Stage' type I'm afraid.

thanks

**Reply by ylazy** [reply]
Thank you :)
